robottelo/virtwho.py

```diff
@@ -0,0 +1,44 @@
+"""Utilities for Virtwho Configure Plugin Testing
+
+All the virtwho-configure UI/CLI/API interfaces should be defined there.
+
+"""
+from robottelo.config import settings
+
+
+def get_hypervisor_type():
```

Comment:
Shouldn't this function take constant name as argument and return specific string? Something like:
```python
def get_hypervisor_type(hypervisor):
    mapping = {
        'esx': 'VMware vSphere / vCenter (esx)',
        # skip for brevity
    }
    return mapping[hypervisor]
```

And then you will use it like that:
```python
hypervisor_type = get_hypervisor_type(hypervisor)
```

Also, I'm not sure about the name here. It doesn't really return hypervisor type, which is already decided in robottelo.properties. We obtain user-visible name based on type.

Reply:
The background is:
The real virt-who config file is defined in /etc/virt-who.d/xxx.conf such as:
[virtwho-config-1]
type=esx
server=xxx.xxx.xxx.xxx
username=xxx
password=xxx
owner=xxx

Virtwho Configure UI change the "esx" type to "VMware vSphere / vCenter (esx)" for user-visible, that means when we select "VMware vSphere / vCenter (esx)" in UI, we have to verify the type is "esx" or not in /etc/virt-who.d/xxx.conf.

Reply:
Another choice is we change the type in robottelo.properties as:
# The hypervisor type, should be one of the values: 
#  'VMware vSphere / vCenter (esx)',
#  'XenServer (xen)',
#  'Red Hat Virtualization Hypervisor (rhevm)',
#  'Microsoft Hyper-V (hyperv)',
#  'libvirt',
#  'Container-native virtualization'
# hypervisor_type=VMware vSphere / vCenter (esx)

Reply:
Oh no, putting UI strings in robottelo.properties doesn't sound like a good design for me. Short names, like `esx`, `xen` etc. are OK.

And if I understand correctly, CLI tests might need to use short names, so you have to do translation between short name and UI string anyway. I think it's better to map from short name to UI string than the other way around.

Reply:
Changed it back and comment as above.